homeassistant/components/climate/mitsubishicontroller.py

```diff
@@ -0,0 +1,159 @@
+"""
+Platform for a Mitsubishi Controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.mitsubishicontroller/
+"""
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import (ClimateDevice, PLATFORM_SCHEMA,
+                                              SUPPORT_TARGET_TEMPERATURE,
+                                              SUPPORT_FAN_MODE,
+                                              SUPPORT_OPERATION_MODE,
+                                              SUPPORT_SWING_MODE)
+from homeassistant.const import CONF_URL, ATTR_TEMPERATURE
+from homeassistant.const import TEMP_FAHRENHEIT
+
+REQUIREMENTS = ['mitsPy==0.1.9']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Setup platform asynchronously."""
+    from mitsPy.manager import Manager
+    controller = Manager(config.get(CONF_URL))
+
+    async def async_register_devices(raw_devices):
+        """Register devices."""
+        async_add_devices(
+            [MitsubishiHvacDevice(
+                device=device,
+                unit_of_measurement=TEMP_FAHRENHEIT) for device in raw_devices]
+        )
+
+    await controller.initialize(async_register_devices)
+
+
+SUPPORT_FLAGS = (
+    SUPPORT_TARGET_TEMPERATURE
+    | SUPPORT_OPERATION_MODE
+    | SUPPORT_FAN_MODE
+    | SUPPORT_SWING_MODE
+)
+
+
+class MitsubishiHvacDevice(ClimateDevice):
+    """Mitsubishi HVAC Device."""
+
+    def __init__(self, device, unit_of_measurement=None,
+                 current_fan_mode=None):
```

Comment:
Why have these keywork arguments? We're only using one of them and always using that one, so it's seems that the last argument can be removed and `unit_of_measurement` be changed to positional argument.